exercism/bracket-push/bracket-push.js

```diff
@@ -0,0 +1,31 @@
+function bracket(bracketString) {
+    if (bracketString.match(/[\{}()[\]]/ig) && bracketString.length%2 === 0){
+      if(bracketString[0] == '{' && bracketString[bracketString.length-1] == '}' && bracketString.match(/[{]/ig).length === bracketString.match(/[}]/ig).length){
+        return true;
+      } else if (bracketString[0] == '[' && bracketString[bracketString.length-1] == ']' && bracketString.match(/[[]/ig).length === bracketString.match(/[\]]/ig).length){
+        return true;
+      } else if (bracketString[0] == '(' && bracketString[bracketString.length-1] == ')' && bracketString.match(/[(]/ig).length === bracketString.match(/[)]/ig).length){
+        return true;
+      } else if (bracketString[0] == '{' &&  bracketString[bracketString.length-1] !== '}'){
+        if (bracketString[1] == '}'){
+          return true;
+        }
+        return false;
+      } else if (bracketString[0] == '[' && bracketString[bracketString.length-1] !== ']'){
+        if (bracketString[1] == ']'){
+          return true;
+        }
+        return false;
+      } else if (bracketString[0] == '(' && bracketString[bracketString.length-1] !== ')'){
+        if (bracketString[1] == ')'){
+          return true;
+        }
+        return false;
+      } else {
+        return false;
+      }
+    }
+    return false;
```

Comment:
Idk if this works or not, but if it does that great.
The code is very hard to read though. My guess is theres a way to refactor this to solve the problem recursively. 
Good job on solving it though! I would suggest cleaning up the code a bit or refactoring to the recursive solution.